### PR TITLE
🔧  Add an option to disable telemetry into config

### PR DESCRIPTION
**Pull request by manfioLP**
now Telemetry can be set on/off based on config flag

if `tap.noTelemetry: true` no Telemetry will run, if `false` (default), telemetry will be sent.

### CODE REVIEW

**Comment by alongir** [comment]
Can config value be more simple, eg: tap.telemetry ?
Default is true. If set to false then it stops.

**Comment by manfioLP** [comment]
> 

it can and I thought of, but after speaking with @Ziul he mentioned it's bad behavior to have a flag=false in Helm deployment

**Comment by manfioLP** [comment]
> Can config value be more simple, eg: tap.telemetry ? Default is true. If set to false then it stops.

it can and I thought of, but after speaking with @Ziul he mentioned it's bad behavior to have a flag=false in Helm deployment

File: cmd/tapRunner.go [hunk 1]
```diff
@@ -106,6 +106,9 @@ func tap() {
 	go watchHubEvents(ctx, kubernetesProvider, cancel)
 	go watchHubPod(ctx, kubernetesProvider, cancel)
 	go watchFrontPod(ctx, kubernetesProvider, cancel)
+	if !config.Config.Tap.NoTelemetry {
+		log.Info().Msg("Starting Telemetry...")
+	}
```
Comment (mertyildiran) [inline cmd/tapRunner.go:111]:
```suggestion
	log.Info().Bool("enabled", !config.Config.Tap.NoTelemetry).Msg("Telemetry")
```

Also print it before this line;

https://github.com/kubeshark/kubeshark/blob/7afb1d8b9b16011431da95e05dfcbee749d33bd8/cmd/tapRunner.go#L80

File: config/config.go [hunk 2]
```diff
@@ -28,6 +28,7 @@ const (
 	FieldNameTag   = "yaml"
 	ReadonlyTag    = "readonly"
 	DebugFlag      = "debug"
+	TelemetryFlag  = "no-telemetry"
```
Comment (mertyildiran) [inline config/config.go:31]: Instead, use the `TELEMETRY_DISABLED` environment variable mentioned in https://github.com/kubeshark/hub/pull/4#discussion_r1305936600

File: helm-chart/templates/04-hub-deployment.yaml [hunk 3]
```diff
@@ -28,6 +28,7 @@ spec:
           command:
             - ./hub
             {{ .Values.tap.debug | ternary "- -debug" "" }}
+            {{ .Values.tap.notelemetry | ternary "- -no-telemetry" "" }}
```
Comment (mertyildiran) [inline helm-chart/templates/04-hub-deployment.yaml:31]: Instead, use the `TELEMETRY_DISABLED` environment variable mentioned in https://github.com/kubeshark/hub/pull/4#discussion_r1305936600

**Review by mertyildiran — CHANGES_REQUESTED** [review]

File: cmd/tapRunner.go (hunk 1 above)
Comment (manfioLP) [reply cmd/tapRunner.go:111]: done

File: config/config.go (hunk 2 above)
Comment (manfioLP) [reply config/config.go:31]: I think it wasn't used at all here, removed.

File: helm-chart/templates/04-hub-deployment.yaml (hunk 3 above)
Comment (manfioLP) [reply helm-chart/templates/04-hub-deployment.yaml:31]: changed to:
```yaml
         command:
            - ./hub
            {{ .Values.tap.debug | ternary "- -debug" "" }}
            - e
            - TELEMETRY_DISABLED={{ .Values.tap.notelemetry }}
```

File: helm-chart/templates/04-hub-deployment.yaml [hunk 4]
```diff
@@ -28,6 +28,8 @@ spec:
           command:
             - ./hub
             {{ .Values.tap.debug | ternary "- -debug" "" }}
+            - e
+            - TELEMETRY_DISABLED={{ .Values.tap.notelemetry }}
```
Comment (mertyildiran) [inline helm-chart/templates/04-hub-deployment.yaml:32]:
```suggestion
```

Add it to here instead:

https://github.com/kubeshark/kubeshark/blob/bf182b633041ebcb975728f41bce86f214328e8f/helm-chart/templates/13-config-map.yaml#L10-L16

**Review by mertyildiran — CHANGES_REQUESTED** [review]

File: helm-chart/templates/04-hub-deployment.yaml (hunk 3 above)
Comment (mertyildiran) [reply helm-chart/templates/04-hub-deployment.yaml]: Since outdated, continuing at https://github.com/kubeshark/kubeshark/pull/1420#discussion_r1309452513

**Comment by mertyildiran** [comment]
Closing because of https://github.com/kubeshark/kubeshark/commit/fa1e7bcf0195f0c5a2e3d523730237c8c380139a

**Pull request closed without merging**